libreyolo/label/server.py

```
@@ -0,0 +1,308 @@
+) -> tuple[ThreadingHTTPServer, str, DatasetSession]:
+    """Open ``data`` and bind the annotator server.
+
+    The dataset is loaded eagerly (so a bad path raises here) and the port is
+    bound eagerly (so an in-use port raises ``OSError`` for the caller to retry).
+    Returns ``(httpd, url, session)``; the caller runs ``serve_forever()``.
+    """
+    session = DatasetSession(data)
+    state = _LabelState(session, device=device, assist=assist)
+    handler = type("BoundLabelHandler", (_Handler,), {"state": state})
+    httpd = ThreadingHTTPServer((host, port), handler)
+    url = "http://%s:%d" % (host, port)
```

Ruff (0.15.15)

[warning] Use format specifiers instead of percent format

(UP031)